components/modal.js

```diff
@@ -99,27 +106,42 @@ const ModalView = ({ visible, children, setVisible, setAddress }) => {
             data={addressArray}
             renderItem={({ item, index }) => (
               <TouchableOpacity
-                onPress={() => {
-                  fetch("http://localhost:8000/api/detail/location", {
-                    method: "POST",
-                    headers: { "Content-Type": "application/json" },
-                    body: JSON.stringify(addressArray[index]),
-                  })
-                    .then((res) => res.json())
-                    .then((json) => {
-                      [
-                        setAddress(json),
-                        localStorage.setItem("address", JSON.stringify(json)),
-                        fetch("http://localhost:8000/api/set/location", {
-                          credentials: "include",
-                          method: "POST",
-                          headers: { "Content-Type": "application/json" },
-                          body: JSON.stringify(json),
-                        }).catch((err) => console.error("error:" + err)),
-                      ];
-                    })
-                    .catch((err) => console.error("error:" + err));
+                // Logic on onPress() fetch calls suggested by Alfredo Sequeida
+                // All the fetch calls for getting the selected address details and then stting up the cookies and fetching popular restaurants
+                onPress={async () => {
+                  const detailData = await fetch(
+                    "http://localhost:8000/api/detail/location",
```

Comment:
So I haven't had a chance to comment on this before as this is the first time I review frontend PRs with API calls. I recall the first one we just pushed to avoid merge conflicts. With that said, it's always best practice to mix UI code with backend API calls. This can get really messy really quickly especially when we use the same API endpoints in multiple files.

So let's start fixing that now to clean up the code a bit. Instead of calling the API endpoints directly in the screens and components directories, let's use the `api` directory at the root of our project, and then in that `api` directory, we want to create files for all of the enpoint roots. For example:

```
api/
├── autocomplete.js
├── db.js
├── get.js
└── search.js
``` 

and for example in `search.js` we can create functions that calls the apis we need and use them in our UIS

```javascript
const search = async (query) => {
   return //the logic to call the search api
}

export {search}
```

and in the UI we can import that function and get the data we need

```jsx
import {search} from "api/search.js";
// use that search functions to get the data
console.log(await search("pizza"));
// or if you want another way of using it
search("pizza").then( (res) => console.log(res) );
```

This will keep the front end A LOT cleaner and it will make making calls to the API calls a lot easier, which we need to do when we add env variables for the domains.